### PR TITLE
[1.6] Various etcd fixes backports

### PR DESCRIPTION
**Pull request by tgraf**
```release-note
Various etcd bug fixes
```

### CODE REVIEW

**Comment by tgraf** [comment]
test-backport-1.6

**Review by aanm — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

File: pkg/kvstore/etcd.go [hunk 1]
```diff
@@ -479,6 +496,7 @@ func (e *etcdClient) renewSession(ctx context.Context) error {
 	log.Infof("Got new lease ID %x", newSession.Lease())
 
 	e.session = newSession
+	e.sessionCancel = sessionCancel
```
Comment (joestringer) [inline pkg/kvstore/etcd.go:499]: The upstream commit here sends `sessionSuccess <- true` a few lines above. Was this intentionally omitted?

File: pkg/kvstore/etcd.go [hunk 2]
```diff
@@ -534,6 +564,7 @@ func (e *etcdClient) renewLockSession(ctx context.Context) error {
 	log.Infof("Got new lock lease ID %x", newSession.Lease())
 
 	e.lockSession = newSession
+	e.lockSessionCancel = sessionCancel
```
Comment (joestringer) [inline pkg/kvstore/etcd.go:567]: Same comment as above here.

File: pkg/kvstore/etcd.go (hunk 1 above)
Comment (tgraf) [reply pkg/kvstore/etcd.go:499]: This was not intentional and was an oversight. I'll fix it up.

Comment (tgraf) [reply pkg/kvstore/etcd.go:499]: https://github.com/cilium/cilium/pull/12753

**Pull request merged**